app/assets/javascripts/galleries/uploader.js

```diff
@@ -26,6 +26,8 @@ function randomString() {
 }
 
 function bindFileInput(fileInput, form, submitButton, formData) {
+  var limit = form.data('limit');
+  var numFiles = 0;
```

Comment:
Should we be subtracting in a `remove` function or when we delete an input line or something? This is a sensible approximation but I can't see a subtraction and users may be confused if they upload a bunch, delete them as a mistake, then re-upload them.